$OpenBSD: patch-platform_x11_x11_main_c,v 1.6 2019/06/25 22:03:46 sthen Exp $

add pledge(2) to mupdf-x11:
  - inet dns : with HAVE_CURL only
  - proc exec : opening an external browser on uri link

Index: platform/x11/x11_main.c
--- platform/x11/x11_main.c.orig
+++ platform/x11/x11_main.c
@@ -7,6 +7,7 @@
 #include <X11/keysym.h>
 #include <X11/XF86keysym.h>
 
+#include <errno.h>
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
@@ -728,7 +729,6 @@ void onselreq(Window requestor, Atom selection, Atom t
 void winreloadpage(pdfapp_t *app)
 {
 	XEvent xev;
-	Display *dpy = XOpenDisplay(NULL);
 
 	xev.xclient.type = ClientMessage;
 	xev.xclient.serial = 0;
@@ -739,8 +739,7 @@ void winreloadpage(pdfapp_t *app)
 	xev.xclient.data.l[0] = 0;
 	xev.xclient.data.l[1] = 0;
 	xev.xclient.data.l[2] = 0;
-	XSendEvent(dpy, xwin, 0, 0, &xev);
-	XCloseDisplay(dpy);
+	XSendEvent(xdpy, xwin, 0, 0, &xev);
 }
 
 void winopenuri(pdfapp_t *app, char *buf)
@@ -927,6 +926,12 @@ int main(int argc, char **argv)
 	tmo_at.tv_sec = 0;
 	tmo_at.tv_usec = 0;
 	timeout = NULL;
+
+	if (pledge("stdio rpath proc exec", NULL) == -1)
+	{
+		fprintf(stderr, "pledge: %s\n", strerror(errno));
+		exit(1);
+	}
 
 	pdfapp_open(&gapp, filename, 0);
 
